### PR TITLE
Allow to specify parent class for active record

### PR DESCRIPTION
**Pull request by morgoth**
Refs #236

POC

### CODE REVIEW

File: lib/good_job/lockable.rb [hunk 1]
```diff
@@ -143,7 +143,7 @@ def with_advisory_lock
       def supports_cte_materialization_specifiers?
         return @_supports_cte_materialization_specifiers if defined?(@_supports_cte_materialization_specifiers)
 
-        @_supports_cte_materialization_specifiers = ActiveRecord::Base.connection.postgresql_version >= 120000
+        @_supports_cte_materialization_specifiers = Job.connection.postgresql_version >= 120000
```
Comment (bensheldon) [inline lib/good_job/lockable.rb:146]: The `GoodJob::Lockable` class is intended to be a generic ActiveRecord mixin, so this hopefully can be:

```suggestion
        @_supports_cte_materialization_specifiers = self.class.connection.postgresql_version >= 120000
```

File: lib/good_job/lockable.rb [hunk 2]
```diff
@@ -158,7 +158,7 @@ def advisory_lock
         WHERE pg_try_advisory_lock(('x'||substr(md5($1 || $2::text), 1, 16))::bit(64)::bigint)
       SQL
       binds = [[nil, self.class.table_name], [nil, send(self.class.primary_key)]]
-      ActiveRecord::Base.connection.exec_query(pg_or_jdbc_query(query), 'GoodJob::Lockable Advisory Lock', binds).any?
+      Job.connection.exec_query(pg_or_jdbc_query(query), 'GoodJob::Lockable Advisory Lock', binds).any?
```
Comment (bensheldon) [inline lib/good_job/lockable.rb:161]: Same here:

```suggestion
      self.class.connection.exec_query(pg_or_jdbc_query(query), 'GoodJob::Lockable Advisory Lock', binds).any?
```

File: lib/good_job/job.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@ module GoodJob
   #
   # Represents a request to perform an +ActiveJob+ job.
   #
-  class Job < ActiveRecord::Base
+  class Job < Object.const_get(GoodJob.active_record_parent_class)
```
Comment (bensheldon) [inline lib/good_job/job.rb:5]: 👍

File: lib/good_job.rb [hunk 4]
```diff
@@ -17,6 +17,9 @@
 #
 # +GoodJob+ is the top-level namespace and exposes configuration attributes.
 module GoodJob
+  # TODO: explain me
```
Comment (bensheldon) [inline lib/good_job.rb:20]:
```suggestion
  # @!attribute [rw] active_record_parent_class
  #   @!scope class
  #   The ActiveRecord parent class inherited by +GoodJob::Job+ (default: +ActiveRecord::Base+).
  #   Use this when using multiple databases or other custom ActiveRecord configuration.
  #   @return [ActiveRecord::Base]
  #   @example Change the base class:
  #     GoodJob.active_record_parent_class = "CustomApplicationRecord"
```

**Review by bensheldon — COMMENTED** [review]
This looks very reasonable 👍

File: lib/good_job/lockable.rb (hunk 1 above)
Comment (morgoth) [reply lib/good_job/lockable.rb]: hmm, this doesn't work somehow

Comment (bensheldon) [reply lib/good_job/lockable.rb]: Whoops. This is within `class_methods` so it just needs to be: `connection.postgresql_version >= 120000`

**Comment by morgoth** [comment]
@bensheldon Anything more I should add to make it merged? I was thinking about some spec, but it might be tricky to test inheritance as the job file is already loaded on spec run.

**Comment by bensheldon** [comment]
@morgoth I agree about the difficulty of testing. It seems like it will either work, or fail across the board. Let's move it forward and see what we learn.

**Comment by bensheldon** [comment]
I forgot that this should also be documented in the Readme. Done in  https://github.com/bensheldon/good_job/commit/9eac0bed6cd4c712d47a1b00e2d5f9224de9a7a1

**Comment by bensheldon** [comment]
This has been released in `v1.9.1`

**Pull request merged**